File: CSharp/Langton/Ant.cs
namespace Langton
{
	public class Ant : IAnt
	{
		public Position Position { get; private set; }

		public Direction Direction { get; private set; }

		public Ant(Direction direction, Position position)
		{
			Direction = direction;
			Position = position;
		}

		public void MoveForward()
		{
			Position = Direction.MoveFrom(Position);
		}

		public void Turn(Rotation rotation)
		{
			Direction = Direction.Rotate(rotation);
		}
	}
}

File: Dart/src/Ant.dart
class Ant {
  Direction direction;
  Cell position;
  
  Ant(Direction this.direction, Cell this.position);
  
  void turn(int rotation) {
    direction = direction.rotate(rotation);
  }
    
  void move() {
    position = direction.moveFrom(position);
  }
}